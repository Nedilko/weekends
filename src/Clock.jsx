function Clock() {
  return (
    <>
      <div className="font-thin uppercase text-center text-slate-600 text-5xl mb-10 mt-20">
        15:03:00
      </div>
    </>
  )
}

export default Clock
