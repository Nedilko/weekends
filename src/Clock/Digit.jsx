import PropTypes from 'prop-types'

Digit.propTypes = {
  digit: PropTypes.string.isRequired,
}

function Digit({ digit }) {
  return <div>{digit}</div>
}

export default Digit
