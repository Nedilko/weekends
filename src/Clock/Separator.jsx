import PropTypes from 'prop-types'

Separator.propTypes = {
  separator: PropTypes.string,
}

function Separator({ separator }) {
  const currentSeparator = separator || ':'
  return <div>{currentSeparator}</div>
}

export default Separator
