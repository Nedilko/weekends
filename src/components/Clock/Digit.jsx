import PropTypes from 'prop-types'

Digit.propTypes = {
  digit: PropTypes.number.isRequired,
}

function Digit({ digit }) {
  return <div className="text-5xl">{digit}</div>
}

export default Digit
