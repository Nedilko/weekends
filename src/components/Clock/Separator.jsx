import PropTypes from 'prop-types'

Separator.propTypes = {
  symbol: PropTypes.string,
}

function Separator({ symbol }) {
  return <div className="text-5xl mt-4">{symbol}</div>
}

export default Separator
