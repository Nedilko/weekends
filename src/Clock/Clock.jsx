import Digit from './Digit'
import Separator from './Separator'

function Clock() {
  return (
    <div className="flex flex-row justify-center font-thin uppercase text-center text-slate-600 text-5xl mb-10 mt-20">
      <Digit digit="15" />
      <Separator />
      <Digit digit="03" />
      <Separator />
      <Digit digit="00" />
    </div>
  )
}

export default Clock
