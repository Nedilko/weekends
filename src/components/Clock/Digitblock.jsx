import PropTypes from 'prop-types'
import Digit from './Digit'
import Separator from './Separator'

Digitblock.propTypes = {
  label: PropTypes.string.isRequired,
  digit: PropTypes.number.isRequired,
  separator: PropTypes.string,
}

function Digitblock({ label, digit, separator }) {
  return (
    <>
      <div className="flex w-24 flex-col">
        <div className="flex h-8 items-center justify-center text-lg font-thin uppercase">
          {label}
        </div>
        <Digit digit={digit} />
      </div>
      {separator && <Separator symbol={separator} />}
    </>
  )
}

export default Digitblock
