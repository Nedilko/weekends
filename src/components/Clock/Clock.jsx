import PropTypes from 'prop-types'
import Digitblock from './Digitblock'

Clock.propTypes = {
  time: PropTypes.shape({
    days: PropTypes.number.isRequired,
    hours: PropTypes.number.isRequired,
    minutes: PropTypes.number.isRequired,
    seconds: PropTypes.number.isRequired,
  }).isRequired,
}

function Clock({ time }) {
  return (
    <div className="flex flex-row mt-10">
      {time.days !== 0 && (
        <Digitblock label="days" digit={time.days} separator=":" />
      )}
      {time.hours !== 0 && (
        <Digitblock label="hours" digit={time.hours} separator=":" />
      )}
      {time.minutes !== 0 && (
        <Digitblock label="minutes" digit={time.minutes} separator=":" />
      )}
      <Digitblock label="seconds" digit={time.seconds} />
    </div>
  )
}

export default Clock
