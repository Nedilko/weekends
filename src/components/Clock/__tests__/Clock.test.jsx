import Clock from '@components/Clock/Clock'
import { render } from '@utils/test-utils'
import { vi } from 'vitest'

vi.mock('@components/Clock/Digit', () => {
  return {
    default: () => <div data-testid="digit"></div>,
  }
})

vi.mock('@components/Clock/Separator', () => {
  return {
    default: () => <div data-testid="separator"></div>,
  }
})

describe('Clock', () => {
  it('should render with all clock numbers and separators', () => {
    const { container } = render(
      <Clock time={{ days: 1, hours: 1, minutes: 1, seconds: 1 }} />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render without days', () => {
    const { container } = render(
      <Clock time={{ days: 0, hours: 1, minutes: 1, seconds: 1 }} />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render without days and hours', () => {
    const { container } = render(
      <Clock time={{ days: 0, hours: 0, minutes: 1, seconds: 1 }} />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render without days, hours and minutes', () => {
    const { container } = render(
      <Clock time={{ days: 0, hours: 0, minutes: 0, seconds: 1 }} />
    )
    expect(container).toMatchSnapshot()
  })
})
