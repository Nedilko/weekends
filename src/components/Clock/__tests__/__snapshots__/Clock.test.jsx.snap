// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock should render with all clock numbers and separators 1`] = `
<div>
  <div
    class="mt-10 flex flex-row justify-around"
  >
    <div
      class="flex w-24 flex-col"
    >
      <div
        class="flex h-8 items-center justify-center text-lg font-thin uppercase"
      >
        days
      </div>
      <div
        data-testid="digit"
      />
    </div>
    <div
      data-testid="separator"
    />
    <div
      class="flex w-24 flex-col"
    >
      <div
        class="flex h-8 items-center justify-center text-lg font-thin uppercase"
      >
        hours
      </div>
      <div
        data-testid="digit"
      />
    </div>
    <div
      data-testid="separator"
    />
    <div
      class="flex w-24 flex-col"
    >
      <div
        class="flex h-8 items-center justify-center text-lg font-thin uppercase"
      >
        minutes
      </div>
      <div
        data-testid="digit"
      />
    </div>
    <div
      data-testid="separator"
    />
    <div
      class="flex w-24 flex-col"
    >
      <div
        class="flex h-8 items-center justify-center text-lg font-thin uppercase"
      >
        seconds
      </div>
      <div
        data-testid="digit"
      />
    </div>
  </div>
</div>
`;

exports[`Clock should render without days 1`] = `
<div>
  <div
    class="mt-10 flex flex-row justify-around"
  >
    <div
      class="flex w-24 flex-col"
    >
      <div
        class="flex h-8 items-center justify-center text-lg font-thin uppercase"
      >
        hours
      </div>
      <div
        data-testid="digit"
      />
    </div>
    <div
      data-testid="separator"
    />
    <div
      class="flex w-24 flex-col"
    >
      <div
        class="flex h-8 items-center justify-center text-lg font-thin uppercase"
      >
        minutes
      </div>
      <div
        data-testid="digit"
      />
    </div>
    <div
      data-testid="separator"
    />
    <div
      class="flex w-24 flex-col"
    >
      <div
        class="flex h-8 items-center justify-center text-lg font-thin uppercase"
      >
        seconds
      </div>
      <div
        data-testid="digit"
      />
    </div>
  </div>
</div>
`;

exports[`Clock should render without days and hours 1`] = `
<div>
  <div
    class="mt-10 flex flex-row justify-around"
  >
    <div
      class="flex w-24 flex-col"
    >
      <div
        class="flex h-8 items-center justify-center text-lg font-thin uppercase"
      >
        minutes
      </div>
      <div
        data-testid="digit"
      />
    </div>
    <div
      data-testid="separator"
    />
    <div
      class="flex w-24 flex-col"
    >
      <div
        class="flex h-8 items-center justify-center text-lg font-thin uppercase"
      >
        seconds
      </div>
      <div
        data-testid="digit"
      />
    </div>
  </div>
</div>
`;

exports[`Clock should render without days, hours and minutes 1`] = `
<div>
  <div
    class="mt-10 flex flex-row justify-around"
  >
    <div
      class="flex w-24 flex-col"
    >
      <div
        class="flex h-8 items-center justify-center text-lg font-thin uppercase"
      >
        seconds
      </div>
      <div
        data-testid="digit"
      />
    </div>
  </div>
</div>
`;
